### PR TITLE
Elasticsearch type error

### PR DESCRIPTION
**Pull request by proteusvacuum**
@esoergel @czue 
Not sure what else to do about this. I couldn't make the `dynamic_template` work properly, so I resorted to this brute force method (after reading pillowtop docs). 

Feel free to push back on this, not completely convinced that it is the best way to go.

http://manage.dimagi.com/default.asp?186943

### CODE REVIEW

File: corehq/apps/users/bulkupload.py [hunk 1]
```diff
@@ -438,8 +438,10 @@ def is_password(password):
                         error = custom_data_validator(data)
                         if error:
                             raise UserUploadError(error)
+                        data = {k: unicode(v) for k, v in data.iteritems()}
                         user.user_data.update(data)
                     if uncategorized_data:
+                        uncategorized_data = {k: unicode(v) for k, v in uncategorized_data.iteritems()}
```
Comment (czue) [inline corehq/apps/users/bulkupload.py]: is this necessary with the other change?

Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: I guess not anymore, will remove

**Comment by czue** [comment]
this looks ok to me - i still find it strange that we don't encounter this with other ES indices, but can address that later

**Comment by czue** [comment]
actually, I don't think we should rebuild the index. the reindex for the unknown users loops runs through all xforms and I think is going to take ages. I'm going to propose that we backlog this ticket and that I fix it when I get to the ES cleanup portion of my ptop work

**Comment by czue** [comment]
(we can merge the pillow changes so that it is fixed moving forwards though)

**Comment by czue** [comment]
I put a don't merge on. Feel free to remove when you revert the index change

File: corehq/pillows/user.py [hunk 2]
```diff
@@ -41,6 +41,24 @@ class UserPillow(AliasedElasticPillow):
...
+        """
+        Make all user_data strings
+        ES doesn't allow dynamic dicts with the same key name to have different types, so coerce everything
+        """
+        user_data = doc_dict.get('user_data', {})
+        if user_data and type(user_data) is dict:
+            try:
+                doc_dict['user_data'] = {k: unicode(v) for k, v in user_data.iteritems()}
+            except UnicodeDecodeError:
+                pass            # If we can't decode it, let elastic deal with it
```
Comment (millerdev) [inline corehq/pillows/user.py]: Would it make sense to convert all values that you can, and only ignore byte strings that can't be converted. Example:

`{u"number": 1, u"bytes": b"\xe4\x94"}`

should be converted to

`{u"number": u"1", u"bytes": b"\xe4\x94"}`

The code here will skip conversion entirely if a single value cannot be converted.

**Pull request merged**